learn/chapter1/2-hello-triangle.md

```diff
@@ -0,0 +1,463 @@
+# Hello Triangle
+
+## The Graphics Pipeline
+
+In OpenGL everything is in 3D space, but the screen and window are a 2D array of pixels so a large part of OpenGL's work is about transforming all 3D coordinates to 2D pixels that fit on your screen. The process of transforming 3D coordinates to 2D pixels is managed by the graphics pipeline of OpenGL. The graphics pipeline can be divided into two large parts: the first transforms your 3D coordinates into 2D coordinates and the second part transforms the 2D coordinates into actual colored pixels. In this tutorial we'll briefly discuss the graphics pipeline and how we can use it to our advantage to create fancy pixels.
+
+The graphics pipeline takes as input a set of 3D coordinates and transforms these to colored 2D pixels on your screen. The graphics pipeline can be divided into several steps where each step requires the output of the previous step as its input. All of these steps are highly specialized (they have one specific function) and can easily be executed in parallel. Because of their parallel nature, graphics cards of today have thousands of small processing cores to quickly process your data within the graphics pipeline by running small programs on the GPU for each step of the pipeline. These small programs are called shaders.
+
+Some of these shaders are configurable by the developer which allows us to write our own shaders to replace the existing default shaders. This gives us much more fine-grained control over specific parts of the pipeline and because they run on the GPU, they can also save us valuable CPU time. Shaders are written in the OpenGL Shading Language (GLSL) and we'll delve more into that in the next tutorial.
+
+Below you'll find an abstract representation of all the stages of the graphics pipeline.
+
+![Pipeline graph](img/2-pipeline.png)
+
+Sections with a blue background are programmable, and the ones with a gray background can be lightly customized using functions. The stages are as follows:
+
+1. Vertex Shader: Vertices are moved into position. This is where things like model position will be applied.
+2. Shape Assembly. OpenGL works by grouping vertices into triangles; this is the stage where that happens.
+3. Geometry Shader: An optional stage of the process. Allows you to fine-tune results from Shape Assembly.
+4. Rasterization: The triangles are converted into fragments.
+5. Fragment Shader: Fragments are modified to include things such as color data. This is where textures and lighting, among other things, are applied.
+6. Tests and Blending: The result of the fragment shader is integrated with the rest of the scene.
+
+That may seem like a lot, but it's quite intuitive once setup is complete and we get into the pipeline.
+
+## Some new functions
+
+We'll need to overload a couple of extra functions to get started. Firstly, we overload OnLoad.
```

Comment:
Should be override. A small nitpick, but might as well do it while you're here.

Author: @devvoid

Reply:
Crud, can't believe I missed that. Fixed in #29